src/_posts/2020-04-28-terraform-with-multiple-environments.md

```diff
@@ -0,0 +1,167 @@
+
+Can I use a single s3 backend with a different profile (a different account) than the rest of the system. Well, yes, the backend can use different credentials than the rest of the setup. In fact, I discovered recently that you can have multiple providers (mixing or multiple accounts of the same one), give them different aliases, and choose for each resource which provider to use.
+
+In fact, some very quick code here how to deal with it, using as a centralized state storage an [Azure](https://azure.microsoft.com/) Blob Storage (mostly because I did not access to the "proper" ones, but I had access to the Codurance Azure playground account).
+
+First the `remote.tf` file, where we indicate where we will store the state for all workspaces
+
+```hcl
+terraform {
+  backend "azurerm" {
+    resource_group_name  = "Jorge-Terraform-Test"
+    storage_account_name = "jorgeterraformtest"
+    container_name       = "terraformstate"
+    key                  = "terraform.tfstate"
+  }
+}
+```
+
+I'm using the `az login` flow for the test. But on actual prod infra you want to use principals.
+
+Then the `variables.tf` file with the only variable that I care about for this quick test
+
+```hcl
+variable "profile" {
+  type=string
+}
+```
+
+And then the main file: `resource.tf`
+
+```
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-1"
+  profile = var.profile
+}
+
+# Create a VPC
+resource "aws_vpc" "example" {
+  cidr_block = "172.35.0.0/16"
+
+  tags = {
+    Name = "jorge's test"
+  }
+}
+```
+
+Finally, for this test I choose to use two separate tfvars files;
+
+`dev.env.tfvars`
+
+```
+profile="codurance-playground"
+```
+
+`prod.env.tfvars`
+
+```
+profile="codurance"
+```
+
+The profile being the one used by terraform to check on my aws credentials file.
+
+
+Finally, to get the ball rolling is the same as with using workspaces on the first section. But now I can handle separate accounts because I am using the single centralized state location.
```

Comment:
I'm not to be quite sure to understand this part `to get the ball rolling is the same as with using workspaces on the first section`

Reply:
I will try to rephrase

Reply:
Check the changes